backend/ibutsu_server/scripts/mongo2postgres.py

```diff
@@ -283,8 +298,72 @@ def migrate_file(collection, Model, vprint, limit=None, filter_=None):
     vprint(" done")
 
 
-def migrate_tables(mongo, vprint, record_limit, month_limit, migrate_files=False):
+def migrate_run(mongo_url, mongo_db, postgres_url, run_id, is_verbose):
+    """Migrate all the data associated with a particular run id"""
+    mongo, _, vprint = setup_objects(mongo_url, mongo_db, postgres_url, is_verbose)
+    # Migrate the actual run first
+    vprint("Migrating run {} ".format(run_id), end="")
+    migrate_table(mongo.runs, Run, vprint, filter_={"_id": run_id})
+    # Then migrate the results
+    vprint("Migrating results ", end="")
+    result_ids = migrate_table(mongo.results, Result, vprint, filter_={"metadata.run": run_id})
+    vprint("Migrating arficats ", end="")
+    migrate_file(
+        GridFSBucket(mongo, "fs"),
+        Artifact,
+        vprint,
+        filter_={"metadata.resultId": {"$in": result_ids}},
+    )
+
+
+def migrate_reports(mongo_url, mongo_db, postgres_url, is_verbose, filter_=None):
+    """Migrate all the reports"""
+    mongo, _, vprint = setup_objects(mongo_url, mongo_db, postgres_url, is_verbose)
+    vprint("Migrating reports ", end="")
+    migrate_table(mongo.reports, Report, vprint)
+    vprint("Migrating files for reports ", end="")
+    migrate_file(GridFSBucket(mongo, "reportFiles"), ReportFile, vprint, filter_=filter_)
+
+
+def migrate_widgets(mongo_url, mongo_db, postgres_url, is_verbose):
+    """Migrate all the widgets for the dashboards"""
+    mongo, _, vprint = setup_objects(mongo_url, mongo_db, postgres_url, is_verbose)
+    vprint("Migrating widgets ", end="")
+    migrate_table(mongo.widgetConfigs, WidgetConfig, vprint)
+
+
+def migrate_projects(mongo, vprint):
+    """Migrate the projects"""
+    vprint("Migrating projects ", end="")
+    for mproject in mongo.projects.find():
+        vprint(".", end="")
+        pproject = Project.query.get(convert_objectid_to_uuid(mproject["_id"]))
```

Comment:
When running locally I got an exception about `context` on this line. I was able to resolve it by using `session.query`.

Reply:
Yeah, cause the models in the server need the application (Flask) context. The code I originally wrote did not use the models from the server.